docs/CodingGuidlines.md

```diff
@@ -0,0 +1,49 @@
+
+# Coding Conventions
+
+Keep it simple. As simple as possible, but not simpler.
+Emphasize readability. Code will be read many times, but only written once.
+
+## History of this document.
+
+* 2018/03/13 Initial version.
+
+
+## Baseline
+
+* These conventions are based on Goggle's guidlines (https://google.github.io/styleguide/cppguide.html). If nothing is stated here to the contrary, refer to the google guidelines.
+* This is a guideline, not a set of commandments. There will be exceptions. Treat them as such.
+* This document is subject to change.
+
+## Attitude
+
+There may be plenty of code violating this convention. Do not use this as justification to deviate from the style guide.
+Instead, use common courtesy and try to leave a place cleaner or at least as clean than you entered it.
+
+
+
+## Visuals
+
+* Spaces instead of Tabs. Use four spaces for normal indentation.
+* Do not indent namespaces. Almost all code is inside a namespace or even nested namespace. Indentation here wastes predious screen realestate.
+
+## Naming Conventions
+
+* Classnames and structs are writting in `CamelCase` (first letter capitalized).
+* Method names are written in `camelCase` (first letter lower case).
+* Member variables are written in `_camelCase`, i.e., lower case first letter and prefixed with an underscore.
```

Comment:
We don't like the underscore thing for every member variable. Maybe it is okay for variables that never ever should be used outside of an class instance. Like: If you use an underscored variable from outside of a class, you are definitely doing something wrong.

Reply:
I am open to other suggestions.
`m_camelCase` or `camelCase` are also popular choices. Directly accessing members of another class is the exception rather than the rule.